Allow operator to specify resource version with command line arg

mitkus opened the pull request:
This adds overrideVersion to KubernetesArgs and a corresponding cmdline argument.

longshorej commented on cli/src/main/scala/com/lightbend/rp/reactivecli/argparse/InputArgs.scala:
```diff
@@ -219,6 +219,11 @@ object InputArgs {
             .text("Sets the cluster IP for Service resources")
             .action(ServiceArgs.set((v, args) => args.copy(clusterIp = Some(v)))),
 
+          opt[String]("override-version")
```
Just `"version"`

longshorej commented on cli/src/main/scala/com/lightbend/rp/reactivecli/annotations/Annotations.scala:
```diff
@@ -73,7 +73,7 @@ case class Annotations(
 object Annotations {
   def apply(labels: Map[String, String], args: GenerateDeploymentArgs): Annotations = {
-    val appVersion = version(labels)
+    val appVersion = version(labels, args)
```
I think we should leave `version` as-is (have it only deal with labels), and add a `version: Option[String]` to `GenerateDeploymentArgs`.

This line (L76) then becomes `val appVersion = args.version.orElse(version(labels))`

longshorej commented on cli/src/main/scala/com/lightbend/rp/reactivecli/annotations/Annotations.scala:
```diff
@@ -147,9 +147,12 @@ object Annotations {
+      case KubernetesArgs(_, _, _, _, _, _, _, _, _, Some(version), _, _, _, _) => Some(version)
+      case _ => labels.get(ns("app-version"))
+    }
+  }
```
As mentioned above, I think we can revert this.

longshorej commented on cli/src/main/scala/com/lightbend/rp/reactivecli/argparse/InputArgs.scala:
```diff
@@ -219,6 +219,11 @@ object InputArgs {
+          opt[String]("override-version")
+            .text("Uses specified version tag for generated resources instead of version in the docker image")
+            .optional()
+            .action(KubernetesArgs.set((v, args) => args.copy(overrideVersion = Some(v)))),
```
Given the above, this will become something like `GenerateDeploymentArgs.set((v, args) => args.copy(version = Some(v)))`

longshorej commented on cli/src/main/scala/com/lightbend/rp/reactivecli/argparse/kubernetes/KubernetesArgs.scala:
```diff
@@ -79,6 +79,7 @@ case class KubernetesArgs(
   transformPodControllers: Option[String] = None,
   transformServices: Option[String] = None,
   namespace: Option[String] = None,
+  overrideVersion: Option[String] = None,
```
I alluded to this above, but I'd rather this be part of `GenerateDeploymentArgs` as it is a general concern. Right now, we only have Kubernetes as a target but the plan is to add DC/OS as well, hence this distinction.

mitkus commented:
Changed structure based on comments, also added a test. The issue with "sbt test" failing to run disappeared miraculously after cleaning everything and building again... From what I've managed to figure out it looks like a bug in scala-native.
